From b24c082d327fd81d61101cc60d89513d3dafdba8 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 17 Aug 2023 16:18:47 +0200
Subject: [PATCH] board: solidrun: imx8mp: fix fdtfile variable setting

Remove unreasonable code from board_late_init that compared hardcoded
"hummingboard-pulse" against "hummingboard-extended" to decide whether
to take carrier name from common structure.

At this point common structure always contains a valid carrier name
already, just construct the fdtfile from som name and carrier name.

Before the rewrite of tlv parsing into common code, HummingBoard Mate
and HummingBoard Ripple were simply treated as HummingBoard Pulse - with
only one device-tree file existing under that name.

After this commit restoring previous method of fdtfile construction,
U-Boot will search for specific DTBs:
- imx8mp-hummingboard-ripple.dtb
- imx8mp-hummingboard-mate.dtb
- imx8mp-hummingboard-pulse.dtb
- imx8mp-hummingboard-extended.dtb
Of these as of today only one exists in Linux, the other 3 won't boot.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c b/board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c
index 140c6c49b8..e4e3745b93 100644
--- a/board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c
+++ b/board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c
@@ -643,14 +643,7 @@ int board_late_init(void)
 
 	// auto-select device-tree
 	if (!env_get("fdtfile")) {
-		
-		char carrier_name[32] = "hummingboard-pulse";
-		// if not extended board
-		if (strcmp(board_id.carrier_name, "hummingboard-extended") != 0) {
-			strcpy(carrier_name, board_id.carrier_name);
-		}
-
-		if(snprintf(fdtfile, sizeof(fdtfile), "%s-%s.dtb", board_id.som_name, carrier_name) >= sizeof(fdtfile)) {
+		if(snprintf(fdtfile, sizeof(fdtfile), "%s-%s.dtb", board_id.som_name, board_id.carrier_name) >= sizeof(fdtfile)) {
 			pr_err("%s: buffer too small, fdtfile truncated!\n", __func__);
 		}
 		env_set("fdtfile", fdtfile);
-- 
2.35.3

